lib/nest-cam-accessory.js

```diff
@@ -56,3 +61,17 @@ const formatMotionDetectionState = function(val) {
     return "not detected";
   }
 };
+
+// --- DoorbellDetectionState ---
+
+const getDoorbellDetectionState = function() {
+  return this.device.last_event &&
+        this.device.last_event.has_person &&
+        !this.device.last_event.end_time;
+};
+
+const formatDoorbellDetectionState = function(val) {
+  if (val) {
+    return "true";//return "0";(Characteristic.ProgrammableSwitchEvent.SINGLE_PRESS = 0;)
```

Comment:
stray comment to clean up?

Reply:
Haven’t managed to test it out yet so this comment in particular is just a reminder for me to try if Characteristic.ProgrammableSwitchEvent doesn’t fire. 

**More importantly, the comment before that should have been a TODO comment, sorry. 

I’m trying to write a line of code that adds a (constant) DoorbellSrv to the NestCamAccessory only if the nest device.long_name contains “Nest Hello”, or simply the word “Hello”. My Nest Hello “long_name” is formatted as “Front Door (Nest Hello)”. 
This is to avoid the other Nest cams populating with HomeKit Doorbell Services.
I believe that’s all that missing from my code. Then I’ll be able to test.

If you can help me write a line of code which helps limit creating a HomeKit Doorbell.Service to only the Nest Hello I can test it asap and report back with results.

Reply:
In the absence for an official Nest API which supports Doorbell Press events, perhaps a rule that limits the Characteristic.ProgrammableSwitchEvent to trigger only when all Nest last_events return a true value. Ex: if “has_sound” + “has_motion” + “has_person” = true, then trigger HomeKit Characteristic.ProgrammableSwitchEvent.

Again, my coding skills are lacking to implement this. So your help is much appreciated.

Reply:
@chrisjshull 
??

Reply:
1. Less user work is best in my opinion. Most Nest Hello users would logically have the words "Nest Hello", "Hello", or the word "Door" as part of the Nest Hello this.device.long_name string.
Please have a look at the changes I've made:
```javascript
if (this.device.long_name = 'Nest Hello') {
    const doorbellSvc = this.addService(Service.Doorbell);
  this.bindCharacteristic(doorbellSvc, Characteristic.ProgrammableSwitchEvent, "Doorbell",
    getDoorbellDetectionState.bind(this), null, formatDoorbellDetectionState.bind(this));
  }
```
is there any way to refine this? 
??
if (this.device.long_name = str.includes("Nest Hello")) {
??
I'm trying to avoid having to add anything to the config.

2. Synthesizing a Doorbell Press Event:
```javascript
return this.device.last_event &&
        this.device.last_event.has_person &&
        this.device.last_event.has_motion &&
        this.device.last_event.has_sound &&
        !this.device.last_event.end_time;
```
Does this seem correct to you? I believe a combination of all last_events, or simply a combination of has_person + has_sound will work. The Nest Hello always makes a "ding dong" sound outside the door where the Nest Hello is placed (there is no way to turn that button press sound off currently). So has_person + has_sound could work to synthesize a Hello button press.

3. Coding for "formatDoorbellDetectionState" done.

Let me know what you think and I'll test this evening. I want to get the coding right before testing so I don't mess up my HomeKit setup by making too many changes to my Homebridge server 😉